install/installer/aws.py

```diff
@@ -0,0 +1,272 @@
+
+            return {
+                'key': login,
+                'secret': password,
+                'created': False,
+            }
+
+    logger.debug(f'Specified keys not belong to user "{user}"')
+    return {}
+
+
+@typechecked
+def create_access_key(
+    iam: boto3.client,
+    user: str,
+) -> Dict[str, Union[str, bool]]:
+    """Create Access Key for exist user.
+
+    Required "iam:CreateAccessKey".
+
+    Args:
+        iam: (boto3.client.IAM) A low-level client representing AWS IAM.
+        user: (str) Cluster.dev user name.
+
+    Returns:
+        Dict[str, Union[str, bool]]:
+        `{'key': 'AWS_ACCESS_KEY_ID', 'secret': 'AWS_SECRET_ACCESS_KEY', 'created': bool}`
+    """
+    response = iam.create_access_key(UserName=user)
+
+    logger.debug('Access keys created')
+    return {
+        'key': response['AccessKey']['AccessKeyId'],
+        'secret': response['AccessKey']['SecretAccessKey'],
+        'created': True,
+    }
+
+
+@typechecked
+def create_user_and_permissions(
+    user: str,
+    login: str,
+    password: str,
+    session: str,
+    release: str,
+) -> Dict[str, Union[str, bool]]:
+    """Create cloud user and attach needed permissions.
+
+    Args:
+        user: (str) Cluster.dev user name.
+        login: (str) Cloud programatic login.
+        password: (str) Cloud programatic password.
+        session: (str) Cloud session token.
+        release: (str) cluster.dev release tag.
+
+    Returns:
+        Dict[str, Union[str, bool]]:
+        `{'key': 'AWS_ACCESS_KEY_ID', 'secret': 'AWS_SECRET_ACCESS_KEY', 'created': bool}`.
+    """
+    iam = boto3.client(
+        'iam',
+        aws_access_key_id=login,
+        aws_secret_access_key=password,
+        aws_session_token=session,
+    )
+
+    # cli_rights_needed_msg
+    msg = (
+        '\n\nRights what you need described here: \n'
+        + f'https://github.com/shalb/cluster.dev/blob/{release}'
+        + '/install/installer_aws_install_req_permissions.json'
+    )
+
+    if user_exist(iam, user, msg):
+        keys = list_user_access_keys(iam, user, msg)
+
+        if not keys:
+            return create_access_key(iam, user)
+
+        creds = check_affiliation_user_and_creds(user, login, password, keys)
+        if creds:
+            return creds
+
+    keys_created = False
+    link = (
+        f'https://github.com/shalb/cluster.dev/blob/{release}'
+        + '/install/installer_aws_install_req_permissions.json'
+    )
+
+    # Create/get policy_arn for user
+    with open('aws_policy.json', 'r') as policy_file:
+        policy = json.dumps(json.load(policy_file))
+
+    try:  # Required "iam:CreatePolicy"
+        response = iam.create_policy(
+            PolicyName='cluster.dev-policy',
+            Path='/cluster.dev/',
+            PolicyDocument=policy,
+            Description='Policy for https://cluster.dev propertly work. '
+            + 'Created by CLI instalator',
```

Comment:
runtime concat of literals is slower
```suggestion
            Description='Policy for https://cluster.dev propertly work. '
            'Created by CLI instalator',
```

Reply:
This script not about optimization. User input and API iteraction - main time eaters

Also, WPS326 Found implicit string concatenation
https://wemake-python-stylegui.de/en/latest/pages/usage/violations/consistency.html#wemake_python_styleguide.violations.consistency.ImplicitStringConcatenationViolation

Reply:
I hate that rule. It's a bit controversial.